### PR TITLE
Add menu bar mode on macOS

### PR DESCRIPTION
**Pull request by vanshg**
Building off of @antonio-ramadas's work in https://github.com/sindresorhus/caprine/pull/231, this PR fixes the Icon Sizes, swaps the incorrect read/unread menu bar icon indicators, renames Continuity to Menu Bar Mode in the View Menu, updates the version number, and merges in all latest changes.

### CODE REVIEW

**Comment by antonio-ramadas** [comment]
@vanshg First of all, thanks for the help! Correct me if I'm wrong, but I believe it is still missing blocking the maximisation of the window when in ```Menu Bar Mode```. [Here](https://github.com/sindresorhus/caprine/pull/231#issuecomment-320387813) I detail about the problem and [here](https://github.com/sindresorhus/caprine/pull/231#issuecomment-320387813) @sindresorhus chose the way to fix it.

**Comment by vanshg** [comment]
@antonio-ramadas Not sure if you meant to, but both links you specified link to the same comment. Regardless, from what I'm trying out right now, the bug seems to be resolved. If you maximize the window, the icon shows up on the Dock, and the menu bar appears and lets you take it out of full screen.

**Comment by antonio-ramadas** [comment]
@vanshg I didn't have the opportunity to run the changes, but if now it is working, then great!
P.S.: You were right, the second link was the same as the first one. I edited the comment to fix it.

**Comment by vanshg** [comment]
PTAL @sindresorhus?

File: config.js [hunk 1]
```diff
@@ -10,6 +10,7 @@ module.exports = new Store({
 			width: 800,
 			height: 600
 		},
+		continuity: false,
```
Comment (sindresorhus) [inline config.js]: This should be called `menuBarMode`

File: index.js [hunk 2]
```diff
@@ -187,6 +193,14 @@ function createMainWindow() {
 	setUserLocale();
 	setUpPrivacyBlocking();
 
+	win.toggle = function () {
```
Comment (sindresorhus) [inline index.js]: Use arrow function

File: index.js [hunk 3]
```diff
@@ -220,14 +231,39 @@ function createMainWindow() {
...
+			app.dock.show();
+		});
+		mainWindow.on('leave-full-screen', () => {
+			if (config.get('continuity')) {
+				app.dock.hide();
+			}
+		});
+	} else {
+		tray.create(mainWindow);
+	}
```
Comment (sindresorhus) [inline index.js]: This whole added block here could be moved into a top-level function that is just called here.

File: index.js [hunk 4]
```diff
@@ -332,3 +368,46 @@ app.on('before-quit', () => {
 		config.set('lastWindowState', mainWindow.getBounds());
 	}
 });
+
+function setContinuity(state) {
```
Comment (sindresorhus) [inline index.js]: Every place `Continuity`, should be changed to `MenuBarMode`

File: index.js [hunk 5]
```diff
@@ -332,3 +368,46 @@ app.on('before-quit', () => {
...
+	if (state) {
+		electron.globalShortcut.register(key, () => {
+			mainWindow.toggle();
+		});
+	} else {
+		electron.globalShortcut.unregister(key);
+	}
+}
+
+function setTrayContinuity(state) {
```
Comment (sindresorhus) [inline index.js]: Too many global functions. I think the whole MenuBarMode functionality could be moved into a class.

File: package.json [hunk 6]
```diff
@@ -1,7 +1,7 @@
 {
-	"name": "caprine",
+  "name": "caprine",
 	"productName": "Caprine",
-	"version": "2.9.0",
+	"version": "2.9.1",
```
Comment (sindresorhus) [inline package.json]: Don't bump this. I'll do so when I'm ready.

File: package.json [hunk 7]
```diff
@@ -1,7 +1,7 @@
 {
-	"name": "caprine",
+  "name": "caprine",
```
Comment (sindresorhus) [inline package.json]: Incorrect indentation.

File: package.json [hunk 8]
```diff
@@ -10,66 +10,76 @@
...
-				}
-			]
-		},
-		"linux": {
-			"target": [
-				"AppImage",
-				"deb"
-			]
-		}
-	}
```
Comment (sindresorhus) [inline package.json]: Same here. Don't do unrelated changes.

File: readme.md [hunk 9]
```diff
@@ -184,3 +193,5 @@ Caprine is a third-party app and is not affiliated with Facebook.
 ## License
 
 MIT
+
+
```
Comment (sindresorhus) [inline readme.md]: Don't do unrelated changes.

File: readme.md [hunk 10]
```diff
@@ -140,7 +147,9 @@ Toggle "Always on Top" | <kbd>Cmd/Ctrl</kbd> <kbd>Shift</kbd> <kbd>t</kbd>
 Toggle window menu     | <kbd>Alt</kbd> *(Windows only)*
 Toggle sidebar         | <kbd>Cmd/Ctrl</kbd> <kbd>Shift</kbd> <kbd>s</kbd>
 Preferences            | <kbd>Cmd/Ctrl</kbd> <kbd>,</kbd>
+Toggle window (global*)| <kbd>Ctrl</kbd> <kbd>m</kbd>
 
+\*In continuity only
```
Comment (sindresorhus) [inline readme.md]: continuity => menu bar mode

File: tray.js [hunk 11]
```diff
@@ -1,30 +1,24 @@
...
-		} else {
-			win.show();
-		}
-	};
-
-	const contextMenu = electron.Menu.buildFromTemplate([
+	contextMenu = electron.Menu.buildFromTemplate([
 		{
-			label: 'Toggle',
+			label: 'Disable Continuity',
```
Comment (sindresorhus) [inline tray.js]: Menu Bar Mode

File: tray.js [hunk 12]
```diff
@@ -1,30 +1,24 @@
 'use strict';
 const path = require('path');
 const electron = require('electron');
+const config = require('./config');
 
 const {app} = electron;
+let contextMenu = null;
 let tray = null;
+let lastCounter = 0;
```
Comment (sindresorhus) [inline tray.js]: `lastCounter` => `previousMessageCount`

File: tray.js [hunk 13]
```diff
@@ -35,18 +29,73 @@ exports.create = win => {
...
+	tray.setToolTip(tip);
+}
+
+function getIconPath(hasUnreadMessages) {
+	let icon = '';
+
+	if (process.platform === 'darwin') {
+		icon = getDarwinIconPath(hasUnreadMessages);
+	} else {
+		icon = getNonDarwinIconPath(hasUnreadMessages);
```
Comment (sindresorhus) [inline tray.js]: Just use a ternary expression here.

File: tray.js [hunk 14]
```diff
@@ -35,18 +29,73 @@ exports.create = win => {
...
+	updateToolTip(messageCount);
+};
+
+exports.destroy = () => {
+	if (tray !== null && !tray.isDestroyed()) {
+		tray.destroy();
+	}
 };
+
+function updateToolTip(counter) {
```
Comment (sindresorhus) [inline tray.js]: `updateToolTip` => `updateTooltip`

File: readme.md [hunk 15]
```diff
@@ -121,6 +121,13 @@ You can send code blocks by using [Markdown syntax](https://github.com/adam-p/ma
 
 When closing the window, the app will continue running in the background, in the dock on macOS and the tray on Linux/Windows. Right-click the dock/tray icon and choose `Quit` to completely quit the app. On macOS, click the dock icon to show the window. On Linux, right-click the tray icon and choose `Toggle` to toggle the window. On Windows, click the tray icon to toggle the window.
 
+If you like to have Caprine minimized on startup, open it from the command-line with the `--minimize` flag.
```
Comment (sindresorhus) [inline readme.md]: This should not be included here.

File: index.js [hunk 16]
```diff
@@ -220,14 +231,39 @@ function createMainWindow() {
...
 }
 
 app.on('ready', () => {
 	const trackingUrlPrefix = `https://l.${domain}/l.php`;
 	electron.Menu.setApplicationMenu(appMenu);
 	mainWindow = createMainWindow();
-	tray.create(mainWindow);
+
+	if (process.platform === 'darwin') {
+		mainWindow.toggleContinuity = function () {
```
Comment (sindresorhus) [inline index.js]: Use arrow function

**Comment by sindresorhus** [comment]
When I select "Disable Continuity" in the tray menu, the app quits.

File: index.js [hunk 17]
```diff
@@ -220,14 +253,22 @@ function createMainWindow() {
 		}
 	});
 
+	win.on('blur', () => {
+		if (process.platform === 'darwin' && config.get('menubarmode') && !win.isFullScreen()) {
```
Comment (sindresorhus) [inline index.js]: `'menubarmode'` should be `'menuBarMode'`. This applies everywhere it's currently used.

File: index.js [hunk 18]
```diff
@@ -187,6 +215,14 @@ function createMainWindow() {
 	setUserLocale();
 	setUpPrivacyBlocking();
 
+	win.toggle = () => {
+		if (this.isVisible()) {
```
Comment (sindresorhus) [inline index.js]: You can't use `this` here. Should be `win`.

File: index.js [hunk 19]
```diff
@@ -131,6 +137,28 @@ function setUpPrivacyBlocking() {
 	});
 }
 
+function setupMenuBarMode(mainWindow) {
+	if (process.platform === 'darwin') {
+		mainWindow.toggleMenuBarMode = function () {
```
Comment (sindresorhus) [inline index.js]: Don't attach methods to the window.

**Comment by sindresorhus** [comment]
And https://github.com/sindresorhus/caprine/pull/374#discussion_r173370112

**Comment by sindresorhus** [comment]
> When I select "Disable Continuity" in the tray menu, the app quits.

This is still an issue.

**Comment by sindresorhus** [comment]
Sorry, but closing for lack of response. Happy to reopen if anyone else wants to continue this and fix the remaining issues.
